lib/matplotlib/backend_bases.py

```diff
@@ -3266,14 +3266,29 @@ def _remove_tool_cbk(self, event):
 
     def _get_image_filename(self, image):
         """Find the image based on its name."""
-        # TODO: better search for images, they are not always in the
-        # datapath
+        if not image:
+            return None
+
         basedir = os.path.join(rcParams['datapath'], 'images')
-        if image is not None:
-            fname = os.path.join(basedir, image)
-        else:
-            fname = None
-        return fname
+        possible_images = [
+            image,
+            image + self.btn_image_extension(),
+            os.path.join(basedir, image) + self.btn_image_extension()]
+
+        for fname in possible_images:
+            if os.path.isfile(fname):
+                return fname
+
+    def btn_image_extension(self):
```

Comment:
"preferred_image_extension"?

Reply:
prone to confusion?

Reply:
It is not actually a preference, it is matter of "does it work?"

Reply:
supported_...

Reply:
could we get away with just having a class-level attribute of `_icon_extension` ?

Reply:
probably would work too...

Reply:
That was my first idea, but I didn't like creating a property with a `raise AttributeError` in the base class.
But i guess you're right, it is alot easier and cleaner for the final implementation of the backend

Reply:
Make it None on the base class which will make it raise a TypeError when fetching the attribute if not overridden?

Reply:
Or default to `'png'`?